### ISSUE TITLE
List materialized views

### ISSUE THREAD

**Issue by aricart** [nats-io/nats-architecture-and-design#154]
## Overview

Go client started listing materialized views like KVs directly. This is useful as it allows the user to list all KVs or ObjectStores returning a value that aligns with the result of `status()` for KV and ObjectStore as provided in those views.  This is syntactic sugar over listing all streams that start with `KV_` or `OBJ_`.

The behavior is documented in ADR-X.

## Clients and Tools

 - [ ] Schemas @ripienaar
 - [ ] CLI @ripienaar
 - [ ] Terraform @ripienaar
 - [ ] Github Actions @ripienaar
 - [ ] Go @piotrpio
 - [ ] Java @scottf
 - [X] JavaScript @aricart https://github.com/nats-io/nats.deno/pull/364
 - [ ] .Net @scottf
 - [ ] C @kozlovic
 - [ ] Python @wallyqs
 - [ ] Ruby @wallyqs
 - [ ] Rust @Jarema @caspervonb
   
## Other Tasks

 - [ ] docs.nats.io updated @jnmoyne
 - [ ] Update ADR to `Implemented`
 - [ ] Update client features spreadsheet

Client authors please update with your progress. If you open issues in your own repositories as a result of this request, please link them to this one by pasting the issue URL in a comment or main issue description.

**Comment by ripienaar** [comment]
@piotrpio looking at the go PR its returning `[]string` but here it says we should be returning equiv of `Status()` which you can do in one API call rather than return the names and then having to load each bucket to get at this.

**Comment by piotrpio** [comment]
@ripienaar was just about to write that. The methods are added to match what we have for streams and consumers in go client, thus there are both available (even though for KV and object store getting names actually has to fetch streams, so it's more of a convenience than performance improvement).

**Comment by ripienaar** [comment]
@piotrpio Yeah, sorry for the back and forth, but the `KeyValueStores()` returns `KeyValue`, to get status would then be another API call for every stream :(

The streams api call return full stream config and state, you can build up 1000s of kv status objects from a single API call.
